doc/api/stream.md

```diff
@@ -443,9 +443,41 @@ first argument. To reliably detect write errors, add a listener for the
 The return value is `true` if the internal buffer does not exceed
 `highWaterMark` configured when the stream was created after admitting `chunk`.
 If `false` is returned, further attempts to write data to the stream should
-stop until the [`'drain'`][] event is emitted. However, the `false` return
-value is only advisory and the writable stream will unconditionally accept and
-buffer `chunk` even if it has not not been allowed to drain.
+stop until the [`'drain'`][] event is emitted.
+
+In case the `false` return value is ignored, the writable stream will
+unconditionally accept and buffer `chunk` even if it has not not been allowed
+to drain. However, Node has a limited amount of memory, and not
+listening to `'drain'` might cause the process to abort by an out of
+memory error, and other types of memory and garbage collection
+issues, such as high RSS.
+
+The `'drain'` event must be handled if the writable is a [Transform][],
+because the `Transform` streams are paused by default until
+they are piped or an `'data'` or `'readable'` event handler is added.
+So, ignoring when `write()` returns false will cause the data to be buffered
+until the stream is flowing, causing a memory leak if it never happens.
+
+A possible way to handle the [`'drain'`][] event to respect
+backpressure is the following:
+
+```js
+function write (data, cb) {
+  if (!stream.write(data)) {
+    stream.once('drain', cb)
+  } else {
+    process.nextTick(cb)
+  }
+}
```

Comment:
Fair enough, and that experience with streams is really valuable. I think you are (quite!) understandably trying to avoid writing a guide on streams and backpressure, while still providing some useful guidance inline in the docs, but I'm concerned that without a little more context people will paste your code into their app, and just move the leak around. No pressure to do a full-scale guide, :-), I'm not -1 on the PR, its an incremental improvement.